ros2_foxglove_bridge/src/message_definition_cache.cpp

```diff
@@ -213,44 +254,61 @@ const MessageSpec& MessageDefinitionCache::load_message_spec(
   }
   const std::string contents{std::istreambuf_iterator(file), {}};
 
-  if (is_action_type) {
-    if (definition_identifier.format == MessageDefinitionFormat::MSG) {
-      const auto [goalDef, resultDef, feedbackDef] = split_action_msg_definition(contents);
-
-      // Define type definitions for each action subtype.
-      // These type definitions may include additional fields such as the goal_id.
-      // See also https://design.ros2.org/articles/actions.html
-      const std::map<std::string, std::string> action_type_definitions = {
-        {ACTION_FEEDBACK_MESSAGE_SUFFIX, "unique_identifier_msgs/UUID goal_id\n" + feedbackDef},
-        {std::string(ACTION_RESULT_SERVICE_SUFFIX) + SERVICE_REQUEST_MESSAGE_SUFFIX,
-         "unique_identifier_msgs/UUID goal_id\n"},
-        {std::string(ACTION_RESULT_SERVICE_SUFFIX) + SERVICE_RESPONSE_MESSAGE_SUFFIX,
-         "int8 status\n" + resultDef},
-        {std::string(ACTION_GOAL_SERVICE_SUFFIX) + SERVICE_REQUEST_MESSAGE_SUFFIX,
-         "unique_identifier_msgs/UUID goal_id\n" + goalDef},
-        {std::string(ACTION_GOAL_SERVICE_SUFFIX) + SERVICE_RESPONSE_MESSAGE_SUFFIX,
-         "bool accepted\nbuiltin_interfaces/msg/Time stamp"}};
-
-      // Create a MessageSpec instance for every action subtype and add it to the cache.
-      for (const auto& [action_suffix, definition] : action_type_definitions) {
-        DefinitionIdentifier definition_id;
-        definition_id.format = definition_identifier.format;
-        definition_id.package_resource_name = package + "/action/" + action_name + action_suffix;
-        msg_specs_by_definition_identifier_.emplace(
-          definition_id, MessageSpec(definition_id.format, definition, package));
-      }
+  if (subfolder == "action") {
+    const auto [goalDef, resultDef, feedbackDef] = split_action_msg_definition(contents);
```

Comment:
Interesting – didn't realize we were already doing a manual split for .action but not .srv. Why were they different?

Reply:
For services it was much easier to use the `_Request` and `_Response` msg definitions which basically didn't need any additional code. For action types, these auto-generated msg definitions didn't exist.